src/litserve/docker_builder.py

```diff
@@ -0,0 +1,118 @@
+# Copyright The Lightning AI team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import logging
+import os
+from pathlib import Path
+
+import warnings
+import litserve as ls
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+logger.propagate = False
+console_handler = logging.StreamHandler()
+console_handler.setLevel(logging.INFO)
+formatter = logging.Formatter("%(asctime)s - %(levelname)s - %(message)s")
+console_handler.setFormatter(formatter)
+logger.addHandler(console_handler)
+
+# COLOR CODES
+RESET = "\u001b[0m"
+RED = "\u001b[31m"
+GREEN = "\u001b[32m"
+BLUE = "\u001b[34m"
+MAGENTA = "\u001b[35m"
+BG_MAGENTA = "\u001b[45m"
+
+# ACTION CODES
+BOLD = "\u001b[1m"
+UNDERLINE = "\u001b[4m"
+INFO = f"{BOLD}{BLUE}[INFO]"
+WARNING = f"{BOLD}{RED}[WARNING]"
+
+
+def color(text, color_code, action_code=None):
+    if action_code:
+        return f"{action_code} {color_code}{text}{RESET}"
+    return f"{color_code}{text}{RESET}"
+
+
+REQUIREMENTS_FILE = "requirements.txt"
+DOCKERFILE_TEMPLATE = """FROM python:3.10-slim
+
+####### Add your own installation commands here #######
+# RUN pip install some-package
+# RUN wget https://path/to/some/data/or/weights
+# RUN apt-get update && apt-get install -y <package-name>
+
+WORKDIR /app
+COPY . /app
+
+# Install litserve and requirements
+RUN pip install --no-cache-dir litserve=={version} {requirements}
+EXPOSE {port}
+CMD ["python", "/app/{server_filename}"]
+"""
+
+# Link our documentation as the bottom of this msg
+SUCCESS_MSG = """
+{BOLD}{MAGENTA}Dockerfile created successfully at{RESET} {UNDERLINE}{dockerfile_path}{RESET}
+
+{BOLD}{BLUE}Follow the instructions below to build & run a docker image:{RESET}
+- To build the Docker image, run: {UNDERLINE}docker build -t <tag> .{RESET}
```

Comment:
Let's rework this so that it's ready to copy-paste:

```
Dockerfile generated, you can now build the container with:
> {UNDERLINE}docker build -t litserve-model .{RESET}

To run the Docker container on the machine:
> {UNDERLINE}docker run -p 8000:8000 litserve-model{RESET}

To push the container to a registry:
> {UNDERLINE}docker push litserve-model{RESET}
```